### PR TITLE
Speed up prime num lookup by binary search.

### PR DESCRIPTION
**Pull request by chuanhong-wang**
Signed-off-by Chuanhong Wang <chuanhong.wang@163.com>

### CODE REVIEW

**Comment by songbaisen** [comment]
lgtm! :+1:  :smile:

File: src/erasure-code/jerasure/ErasureCodeJerasure.cc [hunk 1]
```diff
@@ -158,10 +158,21 @@ bool ErasureCodeJerasure::is_prime(int value)
     151,157,163,167,173,179,
     181,191,193,197,199,211,223,227,229,233,239,241,251,257
   };
-  int i;
-  for (i = 0; i < 55; i++)
-    if (value == prime55[i])
+  int i = sizeof(prime55)/sizeof(int) -1;
```
Comment (tchaikov) [inline src/erasure-code/jerasure/ErasureCodeJerasure.cc]: i'd suggest something like:

```c++
  static const std::array<int, 55> prime55{
    2,3,5,7,11,13,17,19,23,29,31,37,41,43,47,53,59,61,67,71,
    73,79,83,89,97,101,103,107,109,113,127,131,137,139,149,
    151,157,163,167,173,179,
    181,191,193,197,199,211,223,227,229,233,239,241,251,257
  };
  return std::binary_search(std::begin(prime55), std::end(prime55), value);
```

Comment (tchaikov) [inline src/erasure-code/jerasure/ErasureCodeJerasure.cc]: but i don't think `is_prime()` is the concern of performance here, we only check for prime when we are parsing a profile, and normally the number `w` being checked is very small, so i think `std::find()` is more appropriate and faster here in most cases.

Comment (chuanhong-wang) [inline src/erasure-code/jerasure/ErasureCodeJerasure.cc]: In C++ STL header file bits/stl_algo.h, std::binary_search used lower_bound to loockup value, so we can use std::binary_search here.

File: src/erasure-code/jerasure/ErasureCodeJerasure.cc [hunk 2]
```diff
@@ -152,17 +152,14 @@ int ErasureCodeJerasure::decode_chunks(const set<int> &want_to_read,
...
     151,157,163,167,173,179,
     181,191,193,197,199,211,223,227,229,233,239,241,251,257
   };
-  int i;
-  for (i = 0; i < 55; i++)
-    if (value == prime55[i])
-      return true;
-  return false;
+
+  return std::binary_search(std::begin(prime55), std::end(prime55), value);
```
Comment (tchaikov) [inline src/erasure-code/jerasure/ErasureCodeJerasure.cc]: again, i don't think the performance is improved this way.

Comment (chuanhong-wang) [inline src/erasure-code/jerasure/ErasureCodeJerasure.cc]: yes, it doesn't improve performance for most time,  except when user create a new erasure code profile and input a wrong "w".  Should I cancel this pr?

Comment (tchaikov) [inline src/erasure-code/jerasure/ErasureCodeJerasure.cc]: please do so. thanks.

**Comment by dachary** [comment]
@chuanhong-wang this is obviously faster. But the actual gain in overall performance is microscopic and I agree with @tchaikov, this is not worth the effort.

**Comment by chuanhong-wang** [comment]
Cancel the PR, because its actual gain in overall performance is microscopic.

File: src/erasure-code/jerasure/ErasureCodeJerasure.cc (hunk 1 above)
Comment (tchaikov) [inline src/erasure-code/jerasure/ErasureCodeJerasure.cc]: @chuanhong-wang sorry, i was wrong. lower_bound() also do the binary search. the point is 
"the number `w` being checked is very small" so you won't get the performance improvement by using binary search: it takes 7 iterations to find the needle in the haystack if `w == 3`.

**Pull request closed without merging**